src/deploy/functions/deploymentPlanner.ts

```diff
@@ -104,83 +89,44 @@ export function allFunctions(regionMap: RegionMap): CloudFunctionTrigger[] {
-      if (!functionMatchesAnyGroup(fn.name, filters)) {
-        continue;
-      }
-      // Check if this local function has the same name as an exisiting one.
-      const matchingExistingFunction = existingFnsCopy.find((exFn) => exFn.name === fn.name);
-      // Check if the matching exisitng function is scheduled
-      const isMatchingExisitingFnScheduled =
-        matchingExistingFunction?.labels?.["deployment-scheduled"] === "true";
-      // Check if the local function is a scheduled function
-      if (fn.schedule) {
-        // If the local function is scheduled, set its trigger to the correct pubsub topic
-        fn.eventTrigger.resource = getTopicName(fn.name);
-        // and create or update a schedule.
-        regionalDeployment.schedulesToUpsert.push(fn);
-      } else if (isMatchingExisitingFnScheduled) {
-        // If the local function isn't scheduled but the existing one is, delete the schedule.
-        deployment.schedulesToDelete.push(matchingExistingFunction!.name);
-      }
 
-      if (matchingExistingFunction) {
-        // Check if this is an invalid change of trigger type.
-        checkForInvalidChangeOfTrigger(fn, matchingExistingFunction);
-        // Preserve existing environment variables.
-        fn.environmentVariables = {
-          ...matchingExistingFunction.environmentVariables,
-          ...fn.environmentVariables,
-        };
-        regionalDeployment.functionsToUpdate.push(fn);
-        existingFnsCopy = existingFnsCopy.filter((exFn: CloudFunctionTrigger) => {
-          return exFn.name !== fn.name;
-        });
-      } else {
-        regionalDeployment.functionsToCreate.push(fn);
-      }
-    }
-    deployment.regionalDeployments.push(regionalDeployment);
+  const wantRegionalFunctions = functionsByRegion(want.cloudFunctions);
+  const haveRegionalFunctions = functionsByRegion(have.cloudFunctions);
+  for (const region of allRegions(wantRegionalFunctions, haveRegionalFunctions)) {
+    const want = wantRegionalFunctions[region] || [];
+    const have = haveRegionalFunctions[region] || [];
+    deployment.regionalDeployments[region] = calculateRegionalFunctionChanges(want, have, filters);
   }
 
-  // Delete any remaining existing functions that:
-  // 1 - Have the deployment-tool: 'firebase-cli' label and
-  // 2 - Match the --only filters, if any are provided.
-  const functionsToDelete = existingFnsCopy
-    .filter((fn) => {
-      return deploymentTool.isFirebaseManaged(fn.labels);
-    })
-    .filter((fn) => {
-      return functionMatchesAnyGroup(fn.name, filters);
-    });
-  deployment.functionsToDelete = functionsToDelete.map((fn) => {
-    return fn.name;
-  });
-  // Also delete any schedules for functions that we are deleting.
-  for (const fn of functionsToDelete) {
-    if (fn.labels?.["deployment-scheduled"] === "true") {
-      deployment.schedulesToDelete.push(fn.name);
-    }
+  // NOTE(inlined): It's easiest to partition w/ deletes per region and might help
```

Comment:
When originally implementing this, I started with functionsToDelete as a regional concept, and then turned it  into a global one, for a few reasons:

- We don't need to wait for sourceToken to do deletes, so they're not tied to the rest of the function operations in that region
- As you mention, we display the full list of functionsToDelete when asking whether we should delete them.

I think the best reason to split them into regions is potential future features - it would make it very easy to add regional filters, for example.

Reply:
Are you saying I should move deletes to global or that I should remove the comment (or incorporate your feedback into the comment)

Reply:
Sorry, should have been clearer. I'm ok with either implementation. After reviewing the rest of the code the difference is pretty minor - looks like its just a single map().reduce() that you'd be able to get rid of in release.ts if you made functionsToDelete global.